online/src/main/scala/ai/chronon/online/Api.scala

```diff
@@ -190,20 +190,34 @@ abstract class Api(userConf: Map[String, String]) extends Serializable {
     */
   def logResponse(resp: LoggableResponse): Unit
 
+  // override to allow rolling out features/infrastructure changes in a safe, controlled manner
+  def isFeatureFlagEnabled(flagName: String, attributes: java.util.Map[String, String]): Boolean = false
+
   // helper functions
   final def buildFetcher(debug: Boolean = false): Fetcher =
     new Fetcher(genKvStore,
                 Constants.ChrononMetadataKey,
                 logFunc = responseConsumer,
                 debug = debug,
                 externalSourceRegistry = externalRegistry,
-                timeoutMillis = timeoutMillis)
+                timeoutMillis = timeoutMillis,
+                featureFlags = featureFlagBiPredicate)
 
   final def buildJavaFetcher(): JavaFetcher =
-    new JavaFetcher(genKvStore, Constants.ChrononMetadataKey, timeoutMillis, responseConsumer, externalRegistry)
+    new JavaFetcher(genKvStore,
+                    Constants.ChrononMetadataKey,
+                    timeoutMillis,
+                    responseConsumer,
+                    externalRegistry,
+                    featureFlagBiPredicate)
 
   private def responseConsumer: Consumer[LoggableResponse] =
     new Consumer[LoggableResponse] {
       override def accept(t: LoggableResponse): Unit = logResponse(t)
     }
+
+  private def featureFlagBiPredicate: BiPredicate[String, java.util.Map[String, String]] =
```

Comment:
I wonder if having a class called flagStore, the right interface. It might be easier to evolve it in the future.